kong/db/migrations/core/019_320_to_330.lua

```diff
@@ -3,26 +3,29 @@ return {
       up = [[
         CREATE TABLE IF NOT EXISTS "wasm_filter_chains" (
           "id"          UUID                       PRIMARY KEY,
-          "name"        TEXT                       UNIQUE,
           "enabled"     BOOLEAN                    DEFAULT TRUE,
           "route_id"    UUID                       REFERENCES "routes"     ("id") ON DELETE CASCADE,
           "service_id"  UUID                       REFERENCES "services"   ("id") ON DELETE CASCADE,
           "ws_id"       UUID                       REFERENCES "workspaces" ("id") ON DELETE CASCADE,
-          "protocols"   TEXT[],
+          "cache_key"   TEXT                       UNIQUE,
           "filters"     JSONB[],
           "tags"        TEXT[],
           "created_at"  TIMESTAMP WITH TIME ZONE,
-          "updated_at"  TIMESTAMP WITH TIME ZONE
+          "updated_at"  TIMESTAMP WITH TIME ZONE,
+
+          -- service and route are mutually exclusive
+          CONSTRAINT "wasm_filter_chains_scope_ck"
+            CHECK ((route_id IS NULL     AND service_id IS NOT NULL)
+                OR (route_id IS NOT NULL AND service_id IS NULL)
+                OR (route_id IS NULL     AND service_id IS NULL))
```

Comment:
i think `OR (route_id IS NULL     AND service_id IS NULL)` allows global filter chains.

Reply:
Oop, forgot to remove that last case. Thanks!